### DISCUSSION TITLE
tree-sitter init

### DISCUSSION THREAD

**Discussion by AngeloChecked** [tree-sitter/tree-sitter#3848] [Q&A]
Hi everyone,

I’m newbie to tree-sitter, I'm trying to create with my own grammar, but I’m stuck right at the beginning.

According to the documentation here:  
https://tree-sitter.github.io/tree-sitter/creating-parsers  
![image](https://github.com/user-attachments/assets/8a87e126-15a0-4ea5-abcb-c281c3e3f812)

I need to use the command `tree-sitter init` to scaffold the grammar project, but I don’t see that command in the CLI:  
![image](https://github.com/user-attachments/assets/184e2461-e8b9-4afb-9ce0-2b4964e7e451)

Does anyone have any advice?

**Comment by sogaiu** [comment]
~Perhaps this is a case of a discrepancy between the documentation and the actual subcommand name.~

~May be try invoking the `init-config` subcommand?~

_Update_: My bad (^^;

**Comment by WillLillis** [comment]
Which version of `tree-sitter-cli` are you using? It looks like the `init` subcommand was added somewhat recently in https://github.com/tree-sitter/tree-sitter/pull/3694.

(It's likely that you have an older version installed, if you want to use the `init` subcommand I'd update `tree-sitter-cli`)

**Comment by AngeloChecked** [comment]
Strange, I have version `0.24.3`, which should be the latest. Anyway, I manually created the `grammar.js` file, and now the `generate` and `parse` commands work fine!
